src/pages/_app.js

```diff
@@ -80,13 +80,11 @@ const ApplicationLayout = ({ children }) => {
   useChangeLanguage();
   useHandleWindowMessage({
     [WindowMessageTypes.URL_CHANGED]: ({ path }) => {
-      const url = new URL(
+      window.history.pushState(
+        undefined,
+        '',
         `${window.location.protocol}//${window.location.host}${path}`,
       );
-      const hasPage = url.searchParams.has('page');
-      router.push({ pathname: url.pathname, query: {} }, path, {
-        shallow: hasPage,
-      });
```

Comment:
If we don't push this to the router, does the following scenario still work?

- Click on search in the menu
- Click on an event title to go to its detail page
- Click on search in the menu
=> Do you still go to the search page? Because that was a bug in the Vue app that we fixed by pushing to the router.

Reply:
Ah I just tested it and you cannot even get to the detail page of a result now. In other words clicking a link in the iframe doesn't do anything anymore.

That's because we block the link from going through in the Angular app, which was to solve the flickering when we'd push the change to the (Vue) router.

Reply:
See:

- https://github.com/cultuurnet/udb3-angular-app/pull/149
- https://github.com/cultuurnet/udb3-frontend/pull/5

Reply:
You could remove that `e.preventDefault()` in the AngularJS app, but then you might still run into the original issue of the React app not realizing that it's not on the search page anymore, so when you click the search link it doesn't "refresh". But maybe the React router is different enough from the Vue router so that it's not an issue.

Reply:
(I'm using the search page as an example, but it's also an issue on the dashboard, roles, labels, etc pages. Basically everywhere where you click a menu link, click a link in the iframe, then click the same menu link again.)

Reply:
Yeah @bertramakers You're right. the first scenario doesn't work with this method

Reply:
🤔